src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1212,7 +1211,8 @@ public Boolean visitClassType(ClassType t, Type s) {
                 // If t is an intersection, sup might not be a class type
                 if (!sup.hasTag(CLASS)) return isSubtypeNoCapture(sup, s);
                 return sup.tsym == s.tsym
-                    && (t.tsym != s.tsym || t.isReferenceProjection() == s.isReferenceProjection())
+                    && (t.tsym != s.tsym ||
+                        (t.isReferenceProjection() == s.isReferenceProjection() && t.isValueProjection() == s.isValueProjection()))
                      // Check type variable containment
```

Comment:
This check should really be comparing that t.flavor == s.flavor. But this is causing some issues at the moment and will be followed up in JDK-8270882